sponge.go

```diff
@@ -24,13 +27,38 @@ import "sync"
        - economist
    - filter Hacker News if no url
 */
-var itemsToFetch = 10
-var outputLocation = "/tmp/sponge_out.txt"
+
+var defaultOutput = filepath.Join(os.TempDir(), "sponge_out.txt")
+var itemsToFetch = flag.Int("count", 10, "Number of items to fetch from each source")
+var outputLocation = flag.String("out", defaultOutput, "Output file")
 
 type Formatted struct {
 	Body string
 }
 
+func getJsonResponse(url string, v interface{}) error {
+	client := &http.Client{Timeout: 10 * time.Second}
+	resp, err := client.Get(url)
+	if err != nil {
+		return err
+	}
+
+	decoder := json.NewDecoder(resp.Body)
+	if decErr := decoder.Decode(v); decErr != nil {
+		return decErr
+	}
+
+	return nil
+}
+
+func getEnvVar(varName string) (string, error) {
+	envVarValue := os.Getenv(varName)
+	if envVarValue == "" {
+		return "", errors.New(fmt.Sprintf("Env var %s not found", varName))
```

Comment:
Since this is the only error possible from this function, it's not super important to follow this advice but here goes:

usually, you want to declare the variable at the top level so it can be compared against:

```
var ErrEnvVarNotFound = errors.New("Env var not found")
func iSuck() error {
    return ErrEnvVarNotFound
}
```

Of course, this doesn't give you the ability to say *what* env var wasn't found.

In the case where you need to provide additional info, you'll want to create a new type that implements the Error interface but also is able to provide the info you want in the Error() returned string.  Something like

```
type EnvVarError struct {
    varName string
}

func (e EnvVarError) Error() {
    return fmt.Sprintf("Env var %s not found", e.varName)
}

func iProbablySuckToo() error {
    return EnvVarError{"ENV_VAR_NAME"}
}
```

Or just do what you're doing.